### PR TITLE
Build improvements

### PR DESCRIPTION
**Pull request by ngrewe**
Hi,

first of all: Many thanks for your work on the Azure Exporter! We're currently in the process of integrating Azure into our montoring setup, and we have a few improvements to the build pipeline to share. This PR includes the following things:

* introduce a Makefile (inspired by the one used by [coreos/locksmith](https://github.com/coreos/locksmith) that facilitates getting started with the exporter without mucking about with GOPATH etc.
* integrate glide for managing dependency versioning
* Provide a Dockerfile template

There is also one change that might be considered controversial: I've decided to check the vendored dependencies into version control. That's a practice I've seen in quite a few Go projects, and I think it's generally beneficial. Not only does it make getting up to speed with a project dead simple, it also vastly improves the speed of our CI pipelines and is very helpful for ensuring reproducible builds.
Of course, it also results in quite a few more files in the repository, but that cost seems (to me, at least) acceptable.
But if you're unconvinced, I'd be more than happy to remove it again and add a `make vendor` to the CI script instead.

Please let me know what you think!

Best regards,

Niels

### CODE REVIEW

**Comment by brian-brazil** [comment]
We follow the conventions of the Prometheus project, so govendor should be used.

I don't think it's sane to try and deal with developers who don't have a working Go environment.

**Comment by ngrewe** [comment]
Thanks for the feedback! I've updated the PR to use govendor and promu. Please let me know I've you'd like to see further changes.

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,4 @@
 azure-metrics-exporter
 azure.yml
 notes.txt
+/gopath
```
Comment (brian-brazil) [inline .gitignore]: Why would you exclude this?

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# This Makefile has been adapted from the one used by coreos/locksmith.
```
Comment (brian-brazil) [inline Makefile]: This is not the standard Prometheus makefile.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,5 @@
+FROM debian:9-slim
```
Comment (brian-brazil) [inline Dockerfile]: This is not the standard Prometheus dockerfile

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+sudo: false
+language: go
+
+install: true
+
+script:
+ - |
+  if [ ! -d $GOPATH/src/github.com/RobustPerception/azure_metrics_exporter ]; then
```
Comment (brian-brazil) [inline .travis.yml]: It's not a good idea to 2nd guess things like this, presume users have a working setup

Comment (ngrewe) [reply .travis.yml]: I'm afraid I disagree. This is not second-guessing anything. This is being friendly to forks. If you fork the project, Travis CI will check out the project at $GOPATH/src/github.com/GITHUB_USERNAME/azure_metrics_exporter. That means that people forking the project cannot take advantage of having automated builds running against their changes. Linking the checkout to the canonical location restores that functionality and the conditional means that it is a no-op for upstream.

I've addressed your other comments.

Best regards,

Niels

Comment (brian-brazil) [reply .travis.yml]: My point stands. This is non-standard and will result in maintenance burden.

Comment (ngrewe) [reply .travis.yml]: >  This is non-standard and will result in maintenance burden.

Can you please explain why?

Comment (ngrewe) [reply .travis.yml]: I've reviewed the Travis documentation and found a more consise way to achieve this. Would you be happy with that?

**Comment by brian-brazil** [comment]
Thanks!

**Pull request merged**